### PR TITLE
CNV-80440: management: add single alert rule endpoints

### PR DESCRIPTION
**Pull request by sradco**
## Summary

Adds single-rule HTTP endpoints while keeping the existing bulk
APIs from #1047:

- `PATCH /api/v1/alerting/rules/{ruleId}` — labels, drop/restore,
  classification
- `DELETE /api/v1/alerting/rules/{ruleId}` — delete one rule

Single update reuses the same validation and mutation path as
`BulkUpdateAlertRules`. Errors return `ErrorResponse` with an
actionable message (same intent as bulk per-rule `message`).

Preview API is tracked separately in #1180.

### Review feedback addressed

- Reject empty `classification` object (400)
- Filter protected labels (`alertname`, rule ID) on label updates
- Remove redundant platform `alertname` immutability check
- Document bulk delete `ruleIds` limit (1–100) in
  `docs/alert-management.md`

### Tests

- Unit: single update/delete happy paths and 400/401/404/405/413
  edge cases
- E2E: single drop/restore, classification, single delete
- E2E RBAC: anonymous / namespace-scoped / cluster-admin for
  single PATCH and DELETE (parity with create/delete/update bulk)

### Docs

- `docs/alert-management.md` — single vs bulk API matrix and
  response semantics
- `docs/alert-rule-classification.md` — single PATCH response/error
  notes aligned with implementation

## Test plan

- [x] `go test ./pkg/management/... ./internal/managementrouter/...`
- [x] `go build -tags e2e ./test/e2e/...`
- [ ] CI on this PR

Signed-off-by: Shirly Radco <sradco@redhat.com>
Co-authored-by: AI Assistant <noreply@cursor.com>

### CODE REVIEW

File: pkg/management/update_alert_rule_labels.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	sourceRule, err := getOriginalPlatformRuleFromPR(pr, namespace, name, alertRuleId)
+	if err != nil {
+		return "", err
+	}
+
+	userLabels := make(map[string]string, len(sourceRule.Labels))
+	for k, v := range sourceRule.Labels {
+		userLabels[k] = v
+	}
+	for k, pv := range labels {
```
Comment (jgbernalp) [inline pkg/management/update_alert_rule_labels.go]: Do we allow here to replace all the labels without filtering protected labels?

File: pkg/management/update_platform_alert_rule.go [hunk 2]
```diff
@@ -0,0 +1,468 @@
...
+
+func filterAndValidatePlatformLabelChanges(labels map[string]string) (map[string]string, error) {
+	filtered := make(map[string]string)
+	for k, v := range labels {
+		if !isProtectedLabel(k) {
+			filtered[k] = v
+		}
+	}
+	for k, v := range filtered {
+		if k == managementlabels.AlertNameLabel {
```
Comment (jgbernalp) [inline pkg/management/update_platform_alert_rule.go]: I think this was highlighted in a different PR, this is dead code as alert name is part of protected labels

File: pkg/management/update_alert_rule_labels.go (hunk 1 above)
Comment (sradco) [reply pkg/management/update_alert_rule_labels.go]: Fixed. Both paths filter protected labels. User rules merge patches and skip protected keys

File: pkg/management/update_platform_alert_rule.go (hunk 2 above)
Comment (sradco) [reply pkg/management/update_platform_alert_rule.go]: Fixed.

File: internal/managementrouter/alert_rule_update.go [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+}
+
+func validateAlertRuleUpdateFields(f alertRuleUpdateFields) string {
+	if f.AlertingRuleEnabled == nil && f.Labels == nil && f.Classification == nil {
+		return "one of alertingRuleEnabled (toggle drop/restore) or labels (set/unset) or classification is required"
+	}
+	if f.AlertingRuleEnabled != nil && (f.Labels != nil || f.Classification != nil) {
+		return "alertingRuleEnabled cannot be combined with labels or classification in the same request"
+	}
+	return ""
```
[the remark this answers is not in the document]
Comment (sradco) [reply internal/managementrouter/alert_rule_update.go:35]: fixed (alert_rule_update.go:28-34 + tests)

File: test/e2e/single_alert_rule_test.go [hunk 4]
```diff
@@ -0,0 +1,411 @@
...
+
+	ns, cleanup, err := f.CreateUserNamespace(ctx, "test-delete-single")
+	if err != nil {
+		t.Fatalf("Failed to create namespace: %v", err)
+	}
+	defer func() { _ = cleanup() }()
+
+	keepID := mustCreateRule(ctx, t, f, ns, "KeepSingleAlert", "e2e-delete-single-pr")
+	deleteID := mustCreateRule(ctx, t, f, ns, "DeleteSingleAlert", "e2e-delete-single-pr")
+	_ = keepID
```
Comment (jgbernalp) [inline test/e2e/single_alert_rule_test.go:186]: This seems unused, should we use it to assert something post deletion?

Comment (sradco) [reply test/e2e/single_alert_rule_test.go:186]: Updated
